### ISSUE TITLE
`match-description` fails with a `@template` using a complex default value

### ISSUE THREAD

**Issue by Arkellys** [gajus/eslint-plugin-jsdoc#1304]
## Expected behavior
I expect the rule [`match-description`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/match-description.md#readme) do detect correctly the start of the description.

## Actual behavior

The rule [`match-description`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/match-description.md#readme) fails to detect the description on a `@template` using a complex default value. Note that [`require-hyphen-before-param-description`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/require-hyphen-before-param-description.md#readme) fails as well, but I suppose it's linked.

## ESLint Config

```js
"jsdoc/require-hyphen-before-param-description": ["warn", "always", {
  tags: {
    ...
    template: "always"
  }
}],
"jsdoc/match-description": ["warn", {
  contexts: ["any"],
  matchDescription: "^[A-Z].*[^\\.]$",
  tags: {
    ...
    template: {
      message: "@template description should begin with a capital letter and not end with a period.",
      match: true
    }
  }
}]
```
Full config available [here](https://github.com/Arkellys/eslint-config-arklint/blob/main/extensions/jsdoc.js).

## ESLint sample

```js
/**
 * @template {any} T - Arg 1
 * @template {string | number} K - Arg 2
 * @template {any} [R=(K extends keyof T ? T[K] : never)] - Arg 3  ->  Errors here
 * @typedef {any} Test
 */
```

```
There must be a hyphen before @template description.
@template description should begin with a capital letter and not end with a period.
```

 Autofix of  [`require-hyphen-before-param-description`](https://github.com/gajus/eslint-plugin-jsdoc/blob/main/docs/rules/require-hyphen-before-param-description.md#readme) will add the hyphen after `T[K]`.

## Environment

- Node version: 20.9.0
- ESLint version: 8.57.0
- `eslint-plugin-jsdoc` version: 48.10.2

**Comment by brettz9** [comment]
Sorry about that...

**Comment by brettz9** [comment]
In the absence of a proper parser, I've amended @es-joy/jsdoccomment to grab the *last* closing bracket if a starting one is found, and treat that as the end of the "name" portion, with the rest as the "description". That should fix this particular issue, but we should really switch to a proper parser.

**Comment by Arkellys** [comment]
Thank you for the super fast fix!
